### PR TITLE
UAF-2802: Adding liquibase for permissions and responsibilities for PCDM routing plus a few minor changes in labels and validation

### PR DESCRIPTION
**Pull request by nataliac06**

### CODE REVIEW

File: kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+  <changeSet id="UAF-2802-1" author="Natalia Ibanez">
```
Comment (ghost) [inline kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: I am wondering why the permission is in the KFS-SYS namespace and the responsibility is in the KFS-FP namespace. I would have thought both would be in the KFS-FP namespace.

File: kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    <insert tableName="KRIM_ROLE_PERM_T">
+      <column name="ROLE_PERM_ID" valueComputed="KRIM_ROLE_PERM_ID_S.NEXTVAL"/>
+      <column name="PERM_ID" valueComputed="(SELECT PERM_ID FROM KRIM_PERM_T WHERE NM='Initiate Document PCDM' AND OBJ_ID='UAF-2802-1')"/>
+      <column name="ROLE_ID" valueComputed="(SELECT ROLE_ID from KRIM_ROLE_T WHERE ROLE_NM='UA PCard Administrator')"/>
+      <column name="VER_NBR" value="1"/>
+      <column name="OBJ_ID" value="UAF-2802-3"/>
+      <column name="ACTV_IND" value="Y"/>
+    </insert>
+
+    <rollback>
```
Comment (ghost) [inline kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: For the rollback, I think you need to delete the perm attributes first, then the perm. The perm delete would be last.

File: kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        <where>KIM_ATTR_DEFN_ID='13' AND OBJ_ID='UAF-2802-2'</where>
+      </delete>
+      <delete tableName="KRIM_ROLE_PERM_T">
+        <where>OBJ_ID='UAF-2802-3'</where>
+      </delete>
+    </rollback>
+  </changeSet>
+
+  <changeSet id="UAF-2802-2" author="Natalia Ibanez">
+    <preConditions onError="MARK_RAN" onFail="MARK_RAN">
```
Comment (ghost) [inline kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: Shouldn't the preCondition on this changeset be for the responsibility, not the permission?

File: kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      <column name="ATTR_DATA_ID" valueComputed="KRIM_ATTR_DATA_ID_S.NEXTVAL"/>
+      <column name="RSP_ID" valueComputed="(SELECT RSP_ID FROM KRIM_RSP_T WHERE NM='Review PCDM CentralAdministrationReview' AND OBJ_ID='UAF-2802-2')"/>
+      <column name="KIM_TYP_ID" value="7"/>
+      <column name="KIM_ATTR_DEFN_ID" value="40"/>
+      <column name="ATTR_VAL" value="false"/>
+      <column name="VER_NBR" value="1"/>
+      <column name="OBJ_ID" value="UAF-2802-8"/>
+    </insert>
+
+    <rollback>
```
Comment (ghost) [inline kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: For the rollback, I think you need to delete the responsibility attributes first, then the responsibility. The responsibility delete would be last.

**Review by ghost — COMMENTED** [review]
Review complete. Please see my comments.

File: kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml (hunk 2 above)
Comment (nataliac06) [reply kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: Will make the changes for both rollbacks and pre-conditions

File: kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml (hunk 1 above)
Comment (nataliac06) [reply kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: There seem to be inconsistencies all around permissions and roles for PCard. For example the Pcard Admistrator role:UA PCard Administrator is in KFS-PURAP name space. I will ask Misty about this one....

Comment (nataliac06) [reply kfs-core/src/main/resources/edu/arizona/kfs/db/changelog/changesets/db.changelog-UAF-2802.xml]: Julie, I've deleted the liquibase changeset from the pull request and this branch after the conversation with Heather, will put the changes in the dbupgrade scripts.

**Comment by ghost** [comment]
Looks good!

**Pull request merged**